L1Trigger/CSCTriggerPrimitives/src/CSCGEMMatcher.cc

```diff
@@ -39,224 +30,206 @@ void CSCGEMMatcher::setESLookupTables(const CSCL1TPLookupTableME11ILT* conf) { l
-  if (isME1a) {
-    halfStripDiff = std::abs(int(clct.getKeyStrip(2)) - int(cluster.getKeyStripME1a(2)));
-  }
+  bool isLayer2 = false;
 
-  // 98% acceptance cuts
-  unsigned halfStripCut;
-  if (isEven_) {
-    if (isME1a)
-      halfStripCut = maxDeltaHsEvenME1a_;
-    else
-      halfStripCut = maxDeltaHsEven_;
-  } else {
-    if (isME1a)
-      halfStripCut = maxDeltaHsOddME1a_;
-    else
-      halfStripCut = maxDeltaHsOdd_;
-  }
-  // 10 degree chamber is ~0.18 radian wide
-  // 98% acceptance for clusters in odd/even chambers for muons with 5 GeV
-  // {5, 0.02123785, 0.00928431}
-  // This corresponds to 0.12 and 0.052 fractions of the chamber
-  // or 16 and 7 half-strips
-
-  // 20 degree chamber is ~0.35 radian wide
-  // 98% acceptance for clusters in odd/even chambers for muons with 5 GeV
-  // {5, 0.01095490, 0.00631625},
-  // This corresponds to 0.031 and 0.018 fractions of the chamber
-  // or 5 and 3 half-strips
-
-  return halfStripDiff <= halfStripCut;
-}
+  //determine window size
+  unsigned eighthStripCut = isEven_ ? 4 * maxDeltaHsEven_ : 4 * maxDeltaHsOdd_;  // Cut in 1/8 = 4 * cut in 1/2
 
-// match by 1/8-strip
-bool CSCGEMMatcher::matchedClusterLocES(const CSCCLCTDigi& clct, const GEMInternalCluster& cl) const {
-  // key 1/8-strip
-  int key_es = -1;
-
-  //modification of DeltaStrip by CLCT slope
-  int SlopeShift = 0;
-  uint16_t baseSlope = 0;
-  if (mitigateSlopeByCosi_)
-    baseSlope = mitigatedSlopeByConsistency(clct);
-  else
-    baseSlope = clct.getSlope();
-  int clctSlope = pow(-1, clct.getBend()) * baseSlope;
-
-  // for coincidences or single clusters in L1
-  if (cl.isCoincidence() or cl.id().layer() == 1) {
-    key_es = cl.layer1_middle_es();
-    if (station_ == 1 and clct.getKeyStrip() > CSCConstants::MAX_HALF_STRIP_ME1B)
-      key_es = cl.layer1_middle_es_me1a();
-
-    //set SlopeShift for L1 or Copad case
-    SlopeShift =
-        CSCGEMSlopeCorrector(true, clctSlope);  // fixed to facing detectors, must be determined at motherboard level
-  }
-
-  // for single clusters in L2
-  else if (cl.id().layer() == 2) {
-    key_es = cl.layer2_middle_es();
-    if (station_ == 1 and clct.getKeyStrip() > CSCConstants::MAX_HALF_STRIP_ME1B)
-      key_es = cl.layer2_middle_es_me1a();
+  for (const auto& cl : clusters) {
+    // std::cout << "GEM cluster: " << cl << std::endl;
+    // if (!ignoreALCTGEMmatch) std::cout << "IN CLCT-GEM => ALCT-GEM matching L1-L2: " << cl.isMatchingLayer1() << " " << cl.isMatchingLayer2() << std::endl;
+
+    bool isMatchedLayer1 = false;
+    bool isMatchedLayer2 = false;
+
+    if (cl.id1().layer() == 1) {  // cluster has valid layer 1
+      if ((station_ == 1 and enable_match_gem_me1a_ and
+           ((isME1a and cl.roll1() == 8) or (!isME1a and cl.roll1() < 8))) or
+          (station_ == 1 and !enable_match_gem_me1a_ and !isME1a) or (station_ == 2)) {
+        isLayer2 = false;
```

Comment:
`isLayer2` doesn't look to be used anywhere else than in these two if blocks.
To simplify the logic, it could be defined locally here (and in the block below)
```suggestion
        constexpr bool isLayer2 = false;
```